ui/src/utils/react-utils.tsx

```diff
@@ -190,6 +202,50 @@ export function useCloneRef<T>(
     return localRef;
 }
 
+/**
+ * Run an effect at most waitBetweenCalls milliseconds apart, and if there have been changes to the dependencies within
+ * the waiting period, once after that period has passed. If waitBetweenCalls = 0, behaves as useEffect.
+ */
+export function useRateLimitedEffect(
+    effect: EffectCallback,
+    waitBetweenCalls: number,
+    deps?: DependencyList,
+): void {
+    const lastFireTime = useRef<number>();
+    const nextWakeup = useRef<ReturnType<typeof setTimeout>>();
+    const lastDestructor = useRef<void | (() => void)>();
+
+    useEffect(() => {
+        const now = Date.now();
+        if (
+            lastFireTime.current === undefined ||
+            now < lastFireTime.current || // if time turned back, just run the effect and roll with it
```

Comment:
Mahtava kommentti 😄 Se sai minut käymään pienessä kaninkolossa, kun rupesin miettimään onko meillä mitään tilanteita missä aika oikeasti juoksisi "takaperin". Mietinnän jälkeen keksin kaksi keissiä missä näin kävisi:
- Siirtyminen talviaikaan (ei ongelma, timestampeissa taidetaan UTC-aikaa)
- Tietokoneen kellon synkronointi NTP-serverin kanssa. Jos tietokoneen aika on ollut edellä aikaserverin aikaa, niin synkassa aika saattaa hypätä jonkin verran taaksepäin

Käytännössä tämä tuskin lienisi kauhean iso ongelma, mutta JavaScript-maailmassa asia on kuitenkin otettu huomioon. Ajan mittaamiseen on myös Performance-rajapinta (https://developer.mozilla.org/en-US/docs/Web/API/Performance_API/High_precision_timing, esim. https://developer.mozilla.org/en-US/docs/Web/API/Performance/now), joka ainakin myyntipuheiden perusteella väittäisi olevan monotoninen ja järjestelmän kellosta riippumaton. Tässä voisi käyttää tuota `Date.now()`:n sijasta